addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1539,4 +1540,105 @@ options.registry.anchorName = options.Class.extend({
         this.$target.trigger('content_changed');
     },
 });
+
+WysiwygMultizone.include({
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _saveElement: function (outerHTML, recordInfo, editable) {
+        var defs = [this._super.apply(this, arguments)];
+        if ($(editable).data('oe-field') === 'mega_menu_content') {
+            var classesArray = $(editable).attr('class').toString().split(' ');
+            // Remove editable classes from editor before saving classes
+            classesArray.pop('o_editable');
+            classesArray.pop('o_dirty');
+            this._rpc({
+                model: 'website.menu',
+                method: 'save_mega_menu_classes',
```

Comment:
Instead of creating a new method, directly use the 'write' method here... and write on the classes field.